Modules/overlapped.c

```diff
@@ -693,8 +745,27 @@ Overlapped_getresult(OverlappedObject *self, PyObject *args)
             if (transferred != PyBytes_GET_SIZE(self->allocated_buffer) &&
                 _PyBytes_Resize(&self->allocated_buffer, transferred))
                 return NULL;
+
             Py_INCREF(self->allocated_buffer);
             return self->allocated_buffer;
+        case TYPE_READ_FROM:
+            PyBytes_CheckExact(self->read_from.buffer);
```

Comment:
This line doesn't do anything - you need to check the result of `PyBytes_CheckExact` and either assert or fail gracefully.